app/daemon.go

```diff
@@ -151,23 +151,30 @@ func startManager(c *cli.Context) error {
 	// Conversion webhook needs to be started first since we use its port 9501 as readiness port.
 	// longhorn-manager pod becomes ready only when conversion webhook is running.
 	// The services in the longhorn-manager can then start to receive the requests.
-	// Conversion webhook does not longhorn datastore.
+	// Conversion webhook does not need or use longhorn datastore.
 	clientsWithoutDatastore, err := client.NewClients(kubeconfigPath, false, ctx.Done())
 	if err != nil {
 		return err
 	}
+	clients, err := client.NewClients(kubeconfigPath, true, ctx.Done())
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think this change might be problematic. We are everting the fix https://github.com/longhorn/longhorn-manager/pull/2812. AKA, we are trying to start dastore before the conversion webhook which might create a deadlock since the datastore may have to wait for the conversion webhook to be running first 

WDYT?